### ISSUE TITLE
Release wrongly classified as foreign due to ep title

### ISSUE THREAD

**Issue by thezoggy** [Sonarr/Sonarr#861]
release wrongly classified as foreign due to ep title:

![sonarr147](http://zoggy.net/sonarr147.png)

need to update regex/eliminate ep title from that decision.. (use the tvdb data you have. cross reference release regex data)

**Comment by CatmanIX** [comment]
That's actually kinda funny.

**Comment by CatmanIX** [comment]
Also does this when multiple languages are listed, even when english is one of them.

**Comment by thezoggy** [comment]
if any of the blacklisted words are found.. then yes. just because another is in there doesnt exclude it. sonarr knows the ep info.. thus it knows what the ep title is.. so easily should be able to eliminate false positives in that part of the naming.

**Comment by thezoggy** [comment]
just got hit again with this:
`Anthony.Bourdain.Parts.Unknown.S07E03.The.Greek.Islands.HDTV.x264-MiNDTHEGAP`
result is: 'English is wanted, but found Greek'

**Comment by l3uddz** [comment]
Encountered this issue with last nights episode of Below Deck Mediterranean. It was rejected for not being an english release. 

Release name:
Below.Deck.Mediterranean.S01E13.That.Was.Very.Greek.of.Us.HDTV.x264-CRiMSON

**Comment by krin-san** [comment]
I have slightly different issue with Sonarr ignoring RSS-Feed from russian tracker http://www.lostfilm.tv because of incorrect language mapping.

From debug logs below I see release was rejected by 2 reasons: language and quality. I can deal with quality but I can't do anything with language mapping.

RSS feed (part):
```xml
<rss version="0.91">
<channel>
<title>LostFilm.TV</title>
<description>Свежачок от LostFilm.TV</description>
<link>https://www.lostfilm.tv/</link>
<lastBuildDate>Tue, 03 Oct 2017 15:39:29 +0000</lastBuildDate>
<language>ru</language>
<item>
	<title>Бойтесь ходячих мертвецов (Fear the Walking Dead). Эта земля — твоя земля (S03E13) [1080p]</title>
	<category>[1080p]</category>
	<pubDate>Tue, 03 Oct 2017 15:37:46 +0000</pubDate>
	<link>http://tracktor.in/rssdownloader.php?id=24772</link>
</item>
...
</channel>
</rss>
```
Debug log:
```
17-10-3 19:09:01.0|Debug|Parser|Parsing string '   (Fear the Walking Dead).      (S03E13) [1080p]'
17-10-3 19:09:01.0|Debug|Parser|Episode Parsed. (Fear the Walking Dead) - S03E13 
17-10-3 19:09:01.0|Debug|Parser|Language parsed: English
17-10-3 19:09:01.0|Debug|QualityParser|Trying to parse quality for    (Fear the Walking Dead).      (S03E13) [1080p]
17-10-3 19:09:01.0|Debug|Parser|Quality parsed: HDTV-1080p v1
17-10-3 19:09:01.0|Debug|Parser|Release Group parsed: 
17-10-3 19:09:01.0|Debug|AcceptableSizeSpecification|Beginning size check for:    (Fear the Walking Dead).      (S03E13) [1080p]
17-10-3 19:09:01.0|Debug|AcceptableSizeSpecification|Release has unknown size, skipping size check.
17-10-3 19:09:01.0|Debug|LanguageSpecification|Checking if report meets language requirements. English
17-10-3 19:09:01.0|Debug|LanguageSpecification|Report Language: English rejected because it is not wanted, wanted Russian
17-10-3 19:09:01.0|Debug|ReleaseRestrictionsSpecification|Checking if release meets restrictions:    (Fear the Walking Dead).      (S03E13) [1080p]
17-10-3 19:09:01.1|Debug|ReleaseRestrictionsSpecification|[   (Fear the Walking Dead).      (S03E13) [1080p]] No restrictions apply, allowing
17-10-3 19:09:01.1|Debug|QualityAllowedByProfileSpecification|Checking if report meets quality requirements. HDTV-1080p v1
17-10-3 19:09:01.1|Debug|QualityAllowedByProfileSpecification|Quality HDTV-1080p v1 rejected by Series' quality profile
17-10-3 19:09:01.1|Debug|MinimumAgeSpecification|Not checking minimum age requirement for non-usenet report
17-10-3 19:09:01.1|Debug|RetentionSpecification|Not checking retention requirement for non-usenet report
17-10-3 19:09:01.1|Debug|DelaySpecification|Profile does not require a waiting period before download for Torrent.
17-10-3 19:09:01.1|Debug|HistorySpecification|Performing history status check on report
17-10-3 19:09:01.1|Debug|HistorySpecification|Checking current status of episode [2815] in history
17-10-3 19:09:01.1|Debug|DownloadDecisionMaker|Release rejected for the following reasons: [Permanent] Russian is wanted, but found English, [Permanent] HDTV-1080p is not wanted in profile
```

**Comment by markus101** [comment]
@krin-san please open a thread on the forums for support. Your issue is different.

**Comment by speedyflipper** [comment]
Encountered this while trying to download the show Greek. Every episode is thought to be in Greek, not english

**Comment by tedrock** [comment]
Currently occurring with the new show “Russian Doll”

**Comment by Taloth** [comment]
PS, the merged commit a few days ago deals with languages in SxxExx series titles only, not episode titles and not other number formats. Therefore I kept the issue open.

**Comment by Thirrian** [comment]
I'm sure the devs appreciate a pull request to fix this if it's so simple...

**Comment by Rumpa112** [comment]
> I'm sure the devs appreciate a pull request to fix this if it's so simple...

Give tell me what regular-expression engine they use and a list of all the patterns to match I can do it.

**Comment by dhaos21** [comment]
i think this may belong here, This is the cutoff unmet tab, even though the webdl-1080p has been met it is still trying to upgrade the release due to it thinking 2 episodes are Russian and one is Swedish due to the episode name. V3 Sonarr.
![Untitled](https://user-images.githubusercontent.com/3837650/84568278-1fd4b000-ad76-11ea-9f1c-2b214f302289.png)

**Comment by darthShadow** [comment]
Considering the recent changes for fetching resolution from mediainfo, would it not be possible to do the same for the language too?

**Comment by thezoggy** [comment]
language 'tags' vs spoken language vs language of metadata..  dont all align.  then also not everyone uses mediainfo and you need to have said file to process it with mediainfo. so you may think language is x because of filename/release name. but then langauge may be something else via mediainfo.. which do you believe? also multiple audio tracks + subs.. which 'language' do you go off of.

**Comment by Rumpa112** [comment]
> Considering the recent changes for fetching resolution from mediainfo, would it not be possible to do the same for the language too?

You don't even know wtf you are talking about so why vote on my comment!? stupid. The ticket is about releases being classified incorrectly (due to improper regex, google it) and what you are talking about has todo with the language spoken, or rather what language is stored in file metadata,  if any. It might contain several language tag making that useless for this issue.

**Comment by tedrock** [comment]
> You don't even know wtf you are talking about so why vote on my comment!? stupid. The ticket is about releases being classified incorrectly (due to improper regex, google it) and what you are talking about has todo with the language spoken, or rather what language is stored in file metadata, if any. It might contain several language tags making the information useless for this issue.

Considering this is thezoggy’s “ticket” they should know what the ticket is about idiot. Take your attitude elsewhere or make some real contributions.

**Comment by tedrock** [comment]
What’s with the attitude? I subscribe to this ticket for updates, not you ranting and raving about votes and calling someone stupid.

**Comment by Taloth** [comment]
I'll reply to avoid letting it devolve into a rantfest.

Language from mediainfo is indeed not useful in this case. I admit it has some applications (such as discern between audio and sub languages) but not for this ticket.

Rumpa, the regex is not easy. Even for pure scene titles because the same space is used for episode title and language. It's possible to patch some scenarios out but not all of them.   
At this stage in the process the series/episode isn't known yet, so it doesn't have access to the metadata. But ideally one would remove the known episodetitle before trying to parse out the language.   
This is also what we did for the partial fix earlier, removing the text before S07E07 to clear out the series title. That was an important fix because it dealt with scenarios where the entire series would have to be handled manually, rather than individual episodes.

You can see some of the troublesome cases here:
```
My.Series.S07E07.The.Slabside.Redemption.German.Custom.Subbed.720p.HDTV.x264.iNTERNAL-VhV
my.series.s01e07.the.arrow.of.god.french.dvdrip.xvid-pdtvfr
```

This ticket is here to track a low priority issue, it's not on anybodies active todo list, it's simply not important enough. It's squarely in 'not anytime soon, maybe never' territory. So you'll just have to deal with the individual episode manually.

PS: I'm not going to have a discussion about this, got other things to do. But I felt it useful to reiterate where the problem lies.

Have a nice day y'all, enjoy your series!

**Comment by Rumpa112** [comment]
> What’s with the attitude? I subscribe to this ticket for updates, not you ranting and raving about votes (whatever that means) and calling someone stupid. And then all you did was repeat what thezoggy already said but more hostile and less articulate.

Be quiet child.

**Comment by Rumpa112** [comment]
censure FTW fucktards!

**Comment by beyondmeat** [comment]
Why not handle these cases.

Language +"Custom.Subbed" = ignore

For this case
ies.s01e07.the.arrow.of.god.french.dvdrip.xvid-pdtvfr

Select this if there's no other available results. Most releases that are in a different language have it after the rls group now.

**Comment by beechfuzz** [comment]
Just got his with this one, forcing me to manually download the episode:

![image](https://user-images.githubusercontent.com/3413552/94580953-206b5680-0240-11eb-86aa-4f2ef32ff3d9.png)

**Comment by acg5159** [comment]
I have a similar issue, but my issue is that there are multiple language codes for (multi-subtitle files [mkv]) and only one is used.

![image](https://user-images.githubusercontent.com/75958158/195735745-bc344f24-5423-464a-a5aa-5cc5213ae07b.png)

I am looking for ENG subtitles, which this files has and the ENG code is there, but it will not download this version of the file unless I include Italian in the language filter. (Why Italian? ... I think that is the order of languages in you regex parser).

I think the language parser needs to handle more than one language from the filename.

**Comment by GiowGiow** [comment]
> I have a similar issue, but my issue is that there are multiple language codes for (multi-subtitle files [mkv]) and only one is used.
> 
> ![image](https://user-images.githubusercontent.com/75958158/195735745-bc344f24-5423-464a-a5aa-5cc5213ae07b.png)
> 
> I am looking for ENG subtitles, which this files has and the ENG code is there, but it will not download this version of the file unless I include Italian in the language filter. (Why Italian? ... I think that is the order of languages in you regex parser).
> 
> I think the language parser needs to handle more than one language from the filename.
> 
> Here is the snippet from the trace log where it only returns one language: `Language parsed: Italian`
> 
> ```
> 2022-10-13 20:45:40.0|Debug|DownloadDecisionMaker|Processing release '[Erai-raws] Human Bug Daigaku - 02v2 [720p][Multiple Subtitle] [ENG][POR-BR][SPA-LA][SPA][ARA][FRE][GER][ITA][RUS]' from 'Nyaa - Jackett - [Erai-raws]'
> 2022-10-13 20:45:40.0|Debug|Parser|Parsing string '[Erai-raws] Human Bug Daigaku - 02v2 [720p][Multiple Subtitle] [ENG][POR-BR][SPA-LA][SPA][ARA][FRE][GER][ITA][RUS]'
> 2022-10-13 20:45:40.0|Trace|Parser|^\[(?<subgroup>.+?)\][-_. ]?(?<title>[^-]+?)(?:(?<![-_. ]|\b[0]\d+) - )(?:[-_. ]?(?<absoluteepisode>\d{2,3}(\.\d{1,2})?(?!\d+)))+(?:[-_. ]+(?<special>special|ova|ovd))?.*?(?<hash>\[\w{8}\])?(?:$|\.mkv)
> 2022-10-13 20:45:40.0|Debug|Parser|Episode Parsed. Human Bug Daigaku - 002 
> 2022-10-13 20:45:40.0|Debug|Parser|Language parsed: Italian
> 2022-10-13 20:45:40.0|Debug|QualityParser|Trying to parse quality for [Erai-raws] Human Bug Daigaku - 02v2 [720p][Multiple Subtitle] [ENG][POR-BR][SPA-LA][SPA][ARA][FRE][GER][ITA][RUS]
> 2022-10-13 20:45:40.0|Debug|Parser|Quality parsed: HDTV-720p v2
> 2022-10-13 20:45:40.0|Debug|Parser|Release Group parsed: Erai-raws
> 2022-10-13 20:45:40.0|Debug|Parser|Parsing string '[Erai-raws] Human Bug Daigaku - 02v2 [720p][Multiple Subtitle] [ENG][POR-BR][SPA-LA][SPA][ARA][FRE][GER][ITA][RUS]'
> 2022-10-13 20:45:40.0|Debug|SceneMappingService|Found scene mapping for: Human Bug Daigaku. TVDB ID for mapping: 423154
> 2022-10-13 20:45:40.0|Debug|Parser|Parsing string '[Erai-raws] Human Bug Daigaku - 02v2 [720p][Multiple Subtitle] [ENG][POR-BR][SPA-LA][SPA][ARA][FRE][GER][ITA][RUS]'
> 2022-10-13 20:45:40.0|Debug|SceneMappingService|Found scene mapping for: Human Bug Daigaku. TVDB ID for mapping: 423154
> ```
> 
> It should be able to convert `[ENG][POR-BR][SPA-LA][SPA][ARA][FRE][GER][ITA][RUS]` to `English, Portuguese, Spanish, Arabic, French, German, Italian, Russian`

I'm having the same problem, I fully agree that the language parsed should be a set instead of a unique value

**Comment by acg5159** [comment]
> I'm having the same problem, I fully agree that the language parsed should be a set instead of a unique value

Looks like Multiple Language was added in https://github.com/Sonarr/Sonarr/pull/5132, but has not been merged yet.

**Comment by rg9400** [comment]
With the changes made to language profiles in v4, this issue is now critical. Because language profiles are removed, to prevent grabbing dubbed/hardsubbed releases, you need to use a CF that prevents grabbing releases that do not match to the Original Language of the series. However, such a CF would create issues for these episodes because it would parse their language as Spanish, Italian, etc., even if the Original Language of the series is English. This means that any release with the episode title in the release name would match such a "Wrong Language" CF and not be grabbed if such a CF was scored negatively.

These are now fixed on *import* when the audio track language properly updates the language, but the issue is for grabs.

Such a CF is definitely needed in V4 as I am getting many dubbed releases these days, but this issue is preventing the CF from working smoothly.

**Comment by markus101** [comment]
The behaviour is the same as it was before, the solution is still as complicated as it was before as well.
